Gradata/tests/test_emit_pii_redaction.py

```
@@ -0,0 +1,134 @@
+
+
+def test_redactor_failure_aborts_write(tmp_path, monkeypatch):
+    """If _redact_payload raises, emit() must not persist to JSONL or SQLite."""
+    brain = init_brain(tmp_path)
+
+    def _boom(_obj):
+        raise RuntimeError("redactor exploded")
+
+    monkeypatch.setattr(_ev, "_redact_payload", _boom)
+
+    with pytest.raises(Exception):  # EventPersistenceError or the raw RuntimeError
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)